assets/js/atomic/blocks/product-elements/tag-list/constants.tsx

```diff
@@ -4,14 +4,14 @@
 import { __ } from '@wordpress/i18n';
 import { tag, Icon } from '@wordpress/icons';
 
-export const BLOCK_TITLE = __(
+export const BLOCK_TITLE: string = __(
 	'Product Tag List',
 	'woo-gutenberg-products-block'
 );
-export const BLOCK_ICON = (
+export const BLOCK_ICON: JSX.Element = (
 	<Icon icon={ tag } className="wc-block-editor-components-block-icon" />
 );
-export const BLOCK_DESCRIPTION = __(
+export const BLOCK_DESCRIPTION: string = __(
```

Comment:
TS can guess the type of these variables. I think we should rely on Type Inference when possible, what do you think?

Reply:
Thanks for your review, @dinhtungdu. 

> TS can guess the type of these variables. I think we should rely on Type Inference when possible, what do you think?

I think slightly different on that. I believe we should always declare the type, even though TS can guess the type. In the example you mentioned, we explicitly declare that we expect a string. If we would not declare the type and someone would accidentally provide a number or a boolean, the type would automatically change, which is not what we want:

```tsx
export const BLOCK_TITLE = true;          // No error, despite the fact that we expect a string.
export const BLOCK_TITLE: string  = true; // Type 'boolean' is not assignable to type 'string'.
```

By being explicit about the type, we receive errors during development, which I believe, is what we want.

Reply:
> By being explicit about the type, we receive errors during development, which I believe, is what we want.

Makes sense 👍🏽